Requirements.md

```diff
@@ -0,0 +1,120 @@
+   
+## Criteria
+
+A criterion is intended to act as a metric of a 'soft value', such as 'desireability' or 'implementability'. Criteria are especially important when weighing the relative merits of features and in determining the importance that some scenario is satisfied.
+
+1. Prioriterize features used by popular languages.
+   
+   When deciding which features to provide it is important to place emphasis on features that are more likely to be used. We prioriterize support for the languages that are most likely to drive adoption of WASM-GC; this is likely to be the result of a combination of usage of the language and its likelyhood to be used in a WASM-GC engine.
+   
+The 2020 StackOverflow survey reported that the [most popular managed memory languages](https://insights.stackoverflow.com/survey/2020#technology-most-loved-dreaded-and-wanted-languages-loved) -- from the top 20 -- were Typescript, Python, Kotlin, Go, Julia, Dart, C#, Swift, JavaScript, SQL, Shell, HTML/CSS, Scala, Haskell, R, Java, Ruby, PHP. 
+
+   Within these languages, there are languages which are (largely) statically typed -- such as Java and C# -- and languages that are dynamically typed -- such as Python. We prioriterize statically typed languages over dynamically typed language; because those languages are often more performance sensitive, and because support for statically typed languages can often be used for dynamically typed languages.
+   
+1. Security is more important than architectural integrity.
+
+1. Architectural integrity is more important than performance.
+   'Architectural Integrity' is intended to denote simplicity/coherence of design. Generally speaking, architectural integrity can be measured in terms of the number of concepts and the complexity of their relationships.
+      
+   We prioriterize simple, composeable architectures over solution-oriented architectures. This will help to ensure that WASM-GC can evolve and be part of future WASM evolution.
+   
+   Our reasoning here is that performance is often the result of progressive tuning of an implementation. It can typically be improved over time. On the other hand, architectural deficiencies are very difficult to correct.
+      
+1. Performance is more important than convenience.
+
+   Since WASM-GC is not intended as a user-facing technology, convenience here refers to the ease with which toolchain authors can target WASM-GC. 
+   
+   Convenience for toolchain authors is still important however, as that affects adoption of WASM-GC. 
+
+   
+## Critical Success Factors
+
+A critical success factor is an aspect or property of possible solutions that may or may not be directly focused on the primary objective, but none-the-less is estimated to be crucially important to the success of the effort.
+
+1. Permit ‘cycle detection’ between host structures and language structures; so that such cycles can be collected if there are no other references to them.
```

Comment:
Consider rewording as "collection of unreachable cycles".  The point is not so much to detect cycles as to collect them.